IDEAS/Fluid/Taps/BaseClasses/PartialTap.mo

```diff
@@ -0,0 +1,238 @@
+
+protected
+  Modelica.Units.SI.MassFlowRate m_flow_set "DHW mass flow rate at TSet";
+
+public
+  IDEAS.Fluid.Interfaces.IdealSource mFloSouHot(
+    redeclare package Medium = Medium,
+    final control_m_flow=true,
+    allowFlowReversal=false,
+    final control_dp=false)
+    annotation (Placement(transformation(extent={{-16,-10},{4,10}})));
+  Modelica.Fluid.Sources.Boundary_pT bouHot(redeclare package Medium = Medium,
+      nPorts=1) "Sink to model the hot water offtake from the system"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=90,
+        origin={20,-30})));
+  Modelica.Blocks.Sources.RealExpression TCol_in(y=TCol)
+    annotation (Placement(transformation(extent={{30,-70},{50,-50}})));
+  Modelica.Fluid.Sources.Boundary_pT bouCol(
+    redeclare package Medium = Medium,
+    use_T_in=true,
+    nPorts=1) "Supply to model the cold water injection into the system"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=90,
+        origin={60,-30})));
+  IDEAS.Fluid.Interfaces.IdealSource mFloSouCol(
+    redeclare package Medium = Medium,
+    final control_m_flow=true,
+    allowFlowReversal=false,
+    final control_dp=false)
+    annotation (Placement(transformation(extent={{70,-10},{90,10}})));
+  Modelica.Blocks.Sources.RealExpression mFloDis(y=m_flow_set)
+    "DHW mass flow rate if THot < TSet. If the hot water supply temperature is
+    lower than the DHW setpoint temperature, the incoming hot water mass flow 
+    rate to the tap equals the mass flow rate at the setpoint temperature. In 
+    this case, the user will experience DHW discomfort at the tap."
+    annotation (Placement(transformation(extent={{-68,30},{-48,50}})));
+  Modelica.Blocks.Sources.RealExpression mFloCom(y=m_flow_set*(TSet - TCol)/(delTSca.y))
+    "Required incoming hot water mass flow rate based on current THot. If 
+    THot > TSet mixing will occur and cold water will be mixed with the incoming
+    hot water. mFloCom is the required incoming mass flow rate at THot and is 
+    related to mFloSet at TSet via conservation of energy."
+    annotation (Placement(transformation(extent={{-68,10},{-48,30}})));
+  IDEAS.Utilities.Math.SmoothMin mFloHot(deltaX=1e-3*m_flow_nominal)
+    "Hot water mass flow rate. If THot > TSet, mFloHot = mFloCom.
+    If THot < TSet, mFloHot = mFloDis."
+    annotation (Placement(transformation(extent={{-38,20},{-18,40}})));
+  Modelica.Blocks.Sources.RealExpression delT(y=THot.T - TCol) "THot-TCol"
+    annotation (Placement(transformation(extent={{-90,-40},{-70,-20}})));
+  Modelica.Blocks.Sources.RealExpression delT_min(y=0.1)
+    "Minimal value of the temperature difference, to avoid division by zero."
+    annotation (Placement(transformation(extent={{-90,-60},{-70,-40}})));
+  IDEAS.Utilities.Math.SmoothMax delTSca(deltaX=0.1)
+    annotation (Placement(transformation(extent={{-50,-50},{-30,-30}})));
```

Comment:
Can't we use one Smooth function instead of two? The mass flow rate should be `m_flow_set` for `THot<TSet` and linearly decreasing for `THot>TSet`, so I think only `mFloHot` might work, but still to be checked.

Reply:
check for new implementation (only one smooth function)